### ISSUE TITLE
Error running tests

### ISSUE THREAD

**Issue by SimonLab** [healthlocker/oxleas-adhd#252]
![image](https://user-images.githubusercontent.com/6057298/34950796-1559a1f0-fa0c-11e7-8f16-8002de2f5ac5.png)
It looks like the access to the database is blocked by a process (a specific test maybe?) and the other tests are not able to run or finish.

I'm going to create a  new PR to see if Travis has the same issues. If not it might be a configuration problem on my machine.

**Comment by SimonLab** [comment]
Travis is happy which means that it might be an issue with my local config or my database:

![image](https://user-images.githubusercontent.com/6057298/34951826-703f0e04-fa0f-11e7-86ac-394cbe97efc1.png)

We can also clean some unecessary test and some warning:
https://github.com/healthlocker/oxleas-adhd/blob/15942c7deddbcfcabb38165fc16b7ae4cb3fe49d/apps/oxleas_adhd/test/oxleas_adhd_test.exs#L6

```warning: could not find repositories for application :oxleas_adhd.```

Here are some more logs from my tests:
![image](https://user-images.githubusercontent.com/6057298/34951921-d5404ffc-fa0f-11e7-859d-935f76089c73.png)

![image](https://user-images.githubusercontent.com/6057298/34951932-dc3d909e-fa0f-11e7-895a-83e4d4cd017d.png)

**Comment by SimonLab** [comment]
The application run some tests with Wallaby (https://hex.pm/packages/wallaby) which simulate the usre interaction on the browser. Wallaby is using PhantomJS and needs to be install on your machine first.
I think my tests are blocked because PhantomJS is not installed.
I need to run ```npm install phantomjs -g``` but I have a few issues first to fix with npm on my machine (bin/executables are not installed on the correct directory and can't be found on the terminal)

**Comment by SimonLab** [comment]
After installing `phantomjs` as a global package I manage to run all the tests:
![image](https://user-images.githubusercontent.com/6057298/34964311-e0871ec4-fa44-11e7-8ad4-047a8f3509e2.png)

**Comment by reddog** [comment]
@SimonLab Looks like you're having some issues which are may be due to differences in developer machine set up, or as a result of a lack of consistent (Vagrant or other) independent configuration?

Thanks for documenting the steps you're having to take.  If you're having to add in `phantomjs` should that be added to the `package.json` file? https://github.com/healthlocker/oxleas-adhd/blob/master/package.json

In terms of consistent and developer-machine-independent set up, I have started a Vagrant configuration, see https://github.com/healthlocker/oxleas-adhd/pull/241  I am sure that this set up *won't* be complete for your purposes, but maybe something like this would be a way forward?

**Comment by SimonLab** [comment]
thanks for the Vagrant config @reddog .
I don't think the pakcage.json can install npm package globally when running ```npm install```
At the end the issues was more linked to my npm which wasn't configure properly on my machine but I will also add a link to the Healthlocker documentation on the Oxleas Readme which explain that PhantomJS needs to be installed first to be able to run the tests

**Comment by reddog** [comment]
@SimonLab Okay, thanks for that.

If the issues are due to particular set up issues on your machine, would a Vagrant configuration fix this (for you and any/all other developers)?

Having a dependency left out of the set up scripts because it doesn't work on a particular developer's machine (and then relying on developers having fully read the documentation before starting) could lead to other problems down the line maybe?

If this is a bigger issue that can be dealt with right now, should a further issue be opened in order to alert the team project manager that this is a task that could be fixed up to stop other developers requiring time on this issue?

**Comment by reddog** [comment]
@Cleop Thanks for opening the Vagrant issue.

Since @SimonLab's `phantomjs` issue doesn't seem to be down to the set up scripts (as you've said it is installed automatically by `wallaby`), then maybe the Vagrant set up would fix this type of issue.

Would be useful to give some sort of quantification of time that would be saved in implementing a Vagrant configuration?  That would give the project manager a clear way if identifying if time spent fixing issues due to developer machine set up would be better served by implementing a Vagrant development environment?

**Comment by reddog** [comment]
@Cleop Just wanting to make sure we don't loose track of an issue that could bite someone else in the future.  Has the following been done?

> we're going to add a troubleshooting line for this exception in case anyone else encounter
